kbfsgit/runner.go

```diff
@@ -1423,26 +1430,81 @@ func (r *runner) pushAll(ctx context.Context, fs *libfs.FS) (err error) {
 		fmt.Sprintf("Preparing and %s packed refs", verb), "pushprefs")
 }
 
+// parentCommitsForRef returns a map of refs with a list of commits for each
+// ref, newest first. It only includes commits that exist in `localStorer` but
+// not in `remoteStorer`.
+func (r *runner) parentCommitsForRef(ctx context.Context,
+	localStorer gogitstor.Storer, remoteStorer gogitstor.Storer,
+	refs refLookupByName) (commitsByRefName, error) {
```

Comment:
Why is `refs` a map?  The key isn't used that I see.  Could it just be a slice, or is there some uniqueness property that this is important for achieving?

Reply:
1. Uniqueness to ref name.
2. Easier lookup if we ever need to.
3. We need the ref name anyway, and it seems intuitive to me that I use that as the key. On the ticket we're discussing mapping commits to the branch name.

Reply:
Ah, the same src refspec can be used to push into multiple remote branches, so the uniqueness _is_ required I guess.  Carry on.